server/account/src/index.ts

```diff
@@ -288,6 +288,15 @@ export async function signUpJoin (
   return result
 }
 
+/**
+ * @public
+ */
+export async function checkEmail (db: Db, email: string, inviteId: ObjectId): Promise<string> {
+  const invite = await getInvite(db, inviteId)
+  const workspace = await checkInvite(invite, email)
+  return workspace
```

Comment:
I think you should return token | undefined, and if token then store it and redirect